instrumentation/helpers-sql-obfuscation/lib/opentelemetry/instrumentation/helpers/sql_obfuscation.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0module OpenTelemetry
+
+module OpenTelemetry
+  module Instrumentation
+    module Helpers
+      #
+      # This class contains SQL obfuscation behavior to share with
+      # instrumentation for specific database adapters.
+      # The class uses code from: https://github.com/newrelic/newrelic-ruby-agent/blob/1fca78cc7a087421ad58088d8bea72c0362bc62f/lib/new_relic/agent/database/obfuscation_helpers.rb
+      #
+      # To use this in your instrumentation, the `Instrumentation` class for
+      # your gem must contain configuration options for:
+      #  * `:db_statement`
+      #    Example:
+      #    `option :db_statement, default: :include, validate: %I[omit include obfuscate]`
+      #  * `:obfuscation_limit`
+      #    Example:
+      #    `option :obfuscation_limit, default: 2000, validate: :integer`
+      #
+      # If you want to add support for a new adapter, update the following
+      # constants to include keys for your adapter:
+      #  * DIALECT_COMPONENTS
+      #  * CLEANUP_REGEX
+      # You must also add a new constant that uses `generate_regex` with your
+      # adapter's dialect components that is named like
+      # `<ADAPTER>_COMPONENTS_REGEX`, such as: `MYSQL_COMPONENTS_REGEX`.
+      #
+      # @api public
+      class SqlObfuscation
+        # From: https://github.com/newrelic/newrelic-ruby-agent/blob/1fca78cc7a087421ad58088d8bea72c0362bc62f/lib/new_relic/agent/database/obfuscation_helpers.rb
+        COMPONENTS_REGEX_MAP = {
+          single_quotes: /'(?:[^']|'')*?(?:\\'.*|'(?!'))/,
+          double_quotes: /"(?:[^"]|"")*?(?:\\".*|"(?!"))/,
+          dollar_quotes: /(\$(?!\d)[^$]*?\$).*?(?:\1|$)/,
+          uuids: /\{?(?:[0-9a-fA-F]\-*){32}\}?/,
+          numeric_literals: /-?\b(?:[0-9]+\.)?[0-9]+([eE][+-]?[0-9]+)?\b/,
+          boolean_literals: /\b(?:true|false|null)\b/i,
+          hexadecimal_literals: /0x[0-9a-fA-F]+/,
+          comments: /(?:#|--).*?(?=\r|\n|$)/i,
+          multi_line_comments: %r{\/\*(?:[^\/]|\/[^*])*?(?:\*\/|\/\*.*)}
+        }.freeze
+
+        DIALECT_COMPONENTS = {
+          fallback: COMPONENTS_REGEX_MAP.keys,
```

Comment:
Is the idea that this would be something a _future_ instrumentation could refer to as a sort've "catchall" or "default" adapter symbol without needing to update the sql_obfuscation helper gem? I'm fine with that as an idea, but I would maybe want to call it `all` or `default`, and then perhaps make `self.obfuscate_sql` use that as it's default value unless specified. wdyt? Also fine with just dropping this if that's too convoluted.

Reply:
Yes, that's the idea. I love the name change suggestion (`fallback` is the NR name) and using this as the default value.

Do you think I should add a default value for the config hash? Then someone could just pass SQL to the API, though their mileage may vary.

Reply:
Default adapter issue fixed in: 8096551319f684cf850901dbc12100dee43c8694 

Lmk about the config hash. I'm leaning toward no default there so an error is raised if the configs are missing.

Reply:
we should look to avoid raising an error outside of initialization, but i think it would be good to raise an error at init time if possible

Reply:
Instead of raising an error, I decided to do a bit of refactoring so that the whole config hash wasn't required: fd2fa4b0ee585624fb596b823d18d2b475c54f05 

This drops the need for `:db_statement` and provides a default value for `:obfuscation_limit` to make sure the `.obfuscate_sql` method can run with only a SQL statement.